extensions/invite_notify.c

```diff
@@ -0,0 +1,73 @@
+#include <stdinc.h>
+#include <modules.h>
+#include <msgbuf.h>
+#include <client.h>
+#include <hash.h>
+#include <send.h>
+
+static const char inv_notify_desc[] = "Notifies channel on /invite and provides the invite-notify client capability";
+
+static void hook_invite(void *);
+static void m_invited(struct MsgBuf *, struct Client *, struct Client *, int, const char **);
+static unsigned int CAP_INVITE_NOTIFY;
+
+mapi_hfn_list_av1 inv_notify_hfnlist[] = {
+	{ "invite", hook_invite },
+	{ NULL, NULL }
+};
+
+struct Message invited_msgtab = {
+	"INVITED", 0, 0, 0, 0,
+	{mg_ignore, mg_ignore, mg_ignore, mg_ignore, {m_invited, 4}, mg_ignore}
+};
+
+mapi_cap_list_av2 inv_notify_caplist[] = {
+	{ MAPI_CAP_CLIENT, "invite-notify", NULL, &CAP_INVITE_NOTIFY },
+	{ 0, NULL, NULL, NULL }
+};
+
+mapi_clist_av1 inv_notify_clist[] = { &invited_msgtab, NULL };
+
+static void
+invite_notify(struct Client *source, struct Client *target, struct Channel *channel)
+{
+	sendto_channel_local_with_capability(source, CHFL_CHANOP, 0, CAP_INVITE_NOTIFY, channel,
+		":%s NOTICE %s :%s is inviting %s to %s.",
+		me.name, channel->chname, source->name, target->name, channel->chname);
+	sendto_channel_local_with_capability(source, CHFL_CHANOP, CAP_INVITE_NOTIFY, 0, channel,
+		":%s!%s@%s INVITE %s %s", source->name, source->username,
+		source->host, target->name, channel->chname);
+}
+
+static void
+hook_invite(void *data_)
+{
+	hook_data_channel_approval *data = data_;
+
+	if (!data->approved)
+		/* Don't notify if another hook is rejecting the invite.
+		 * This won't work if the other hook is after us... but really, it's
+		 * the thought that counts.
+		 */
+		return;
+
+	sendto_server(NULL, NULL, 0, 0, "ENCAP * INVITED %s %s %s",
```

Comment:
Doesn't the third argument here need to be `CAP_TS6 | CAP_ENCAP` ?

Reply:
yes